crates/host/src/wasmbus/lattice.rs

```diff
@@ -0,0 +1,2954 @@
+    max_execution_time: Duration,
+    runtime: Runtime,
+    event_rx: mpsc::Sender<(String, String, serde_json::Value)>,
+    ctl_topic_prefix: String,
+    heartbeat_abort: AbortHandle,
+}
+
+impl Lattice {
+    /// Instantiate a new Lattice which is capable of running workloads for a particular wasmCloud
+    /// lattice.
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) async fn new(
+        name: String,
+        client: async_nats::Client,
+        ctl_client: async_nats::Client,
+        jetstream_client: async_nats::jetstream::Context,
+        host_key: String,
+        labels: Arc<RwLock<BTreeMap<String, String>>>,
+        metrics: Arc<HostMetrics>,
+        max_execution_time: Duration,
+        runtime: Runtime,
+        event_rx: mpsc::Sender<(String, String, serde_json::Value)>,
+        config: LatticeConfig,
+        heartbeat_interval: Interval,
+        host_token: jwt::Token<jwt::Host>,
+    ) -> anyhow::Result<Arc<Self>> {
+        let (stop_tx, stop_rx) = watch::channel(None);
+
+        let bucket = format!("LATTICEDATA_{}", name.clone());
+        let data = create_bucket(&jetstream_client, &bucket).await?;
+
+        let config_bucket = format!("CONFIGDATA_{}", name.clone());
+        let config_data = create_bucket(&jetstream_client, &config_bucket).await?;
+
+        let supplemental_config = if config.config_service_enabled {
+            load_supplemental_config(&client, &name, &labels.read().await.clone()).await?
+        } else {
+            SupplementalConfig::default()
+        };
+
+        let registry_config = RwLock::new(supplemental_config.registry_config.unwrap_or_default());
+        merge_registry_config(&registry_config, config.oci_opts.clone()).await;
+
+        let policy_manager = PolicyManager::new(
+            client.clone(),
+            PolicyHostInfo {
+                public_key: host_key.clone(),
+                lattice: name.clone(),
+                labels: HashMap::from_iter(labels.read().await.clone()),
+            },
+            config.policy_service_config.policy_topic.clone(),
+            config.policy_service_config.policy_timeout_ms,
+            config.policy_service_config.policy_changes_topic.clone(),
+        )
+        .await?;
+
+        let secrets_manager = Arc::new(SecretsManager::new(
+            &config_data,
+            config.secrets_topic_prefix.as_ref(),
+            &client,
+        ));
+        let config_generator = BundleGenerator::new(config_data.clone());
+
+        let (data_watch_abort, data_watch_abort_reg) = AbortHandle::new_pair();
+        let (queue_abort, queue_abort_reg) = AbortHandle::new_pair();
+        let (heartbeat_abort, heartbeat_abort_reg) = AbortHandle::new_pair();
+        let lattice = Self {
+            name: Arc::from(name.clone()),
+            labels,
+            // TODO this is in theory redundant with some other config options
+            ctl_topic_prefix: config.ctl_topic_prefix.clone(),
```

Comment:
Yeah do you even need to know this topic prefix in this struct, or are the connections enough?